vm/src/stdlib/os.rs

```diff
@@ -925,6 +927,102 @@ fn os_chdir(path: PyStringRef, vm: &VirtualMachine) -> PyResult<()> {
     env::set_current_dir(path.as_str()).map_err(|err| convert_io_error(vm, err))
 }
 
+#[cfg(unix)]
+fn os_get_inheritable(fd: RawFd, vm: &VirtualMachine) -> PyResult<bool> {
+    use nix::fcntl::fcntl;
+    use nix::fcntl::FcntlArg;
+    let flags = fcntl(fd, FcntlArg::F_GETFD);
+    match flags {
+        Ok(ret) => Ok((ret & libc::FD_CLOEXEC) == 0),
+        Err(err) => Err(convert_nix_error(vm, err)),
+    }
+}
+
+#[cfg(unix)]
+fn os_set_inheritable(fd: RawFd, inheritable: bool, vm: &VirtualMachine) -> PyResult<()> {
+    let _set_flag = || {
+        use nix::fcntl::fcntl;
+        use nix::fcntl::FcntlArg;
+        use nix::fcntl::FdFlag;
+
+        let flags = FdFlag::from_bits_truncate(fcntl(fd, FcntlArg::F_GETFD)?);
+        let mut new_flags = flags;
+        new_flags.set(FdFlag::from_bits_truncate(libc::FD_CLOEXEC), !inheritable);
+        if flags != new_flags {
+            fcntl(fd, FcntlArg::F_SETFD(new_flags))?;
+        }
+        Ok(())
+    };
+    _set_flag().or_else(|err| Err(convert_nix_error(vm, err)))
+}
+
+#[cfg(unix)]
+fn os_get_blocking(fd: RawFd, vm: &VirtualMachine) -> PyResult<bool> {
+    use nix::fcntl::fcntl;
+    use nix::fcntl::FcntlArg;
+    let flags = fcntl(fd, FcntlArg::F_GETFL);
+    match flags {
+        Ok(ret) => Ok((ret & libc::O_NONBLOCK) == 0),
+        Err(err) => Err(convert_nix_error(vm, err)),
+    }
+}
+
+#[cfg(unix)]
+fn os_set_blocking(fd: RawFd, blocking: bool, vm: &VirtualMachine) -> PyResult<()> {
+    let _set_flag = || {
+        use nix::fcntl::fcntl;
+        use nix::fcntl::FcntlArg;
+        use nix::fcntl::OFlag;
+
+        let flags = OFlag::from_bits_truncate(fcntl(fd, FcntlArg::F_GETFL)?);
+        let mut new_flags = flags;
+        new_flags.set(OFlag::from_bits_truncate(libc::O_NONBLOCK), !blocking);
+        if flags != new_flags {
+            fcntl(fd, FcntlArg::F_SETFL(new_flags))?;
+        }
+        Ok(())
+    };
+    _set_flag().or_else(|err| Err(convert_nix_error(vm, err)))
+}
+
+#[cfg(unix)]
+fn os_pipe(vm: &VirtualMachine) -> PyResult<(RawFd, RawFd)> {
+    use nix::unistd::close;
+    use nix::unistd::pipe;
+    match pipe() {
+        Err(err) => Err(convert_nix_error(vm, err)),
```

Comment:
I think you could do something like `let (rfd, wfd) = pipe().map_err(|err| ...)?;`